src/shaders/line.vertex.glsl

```diff
@@ -118,7 +121,8 @@ void main() {
     v_uv = vec4(a_uv_x, a_split_index * texel_height - half_texel_height, a_clip_start, a_clip_end);
 #else
     v_uv = vec4(a_uv_x, 0.0, a_clip_start, a_clip_end);
-#endif
+#endif // #ifdef RENDER_LINE_GRADIENT
```

Comment:
What about if `RENDER_LINE_TRIM_OFFSET` is not defined but we have `RENDER_LINE_GRADIENT` ? Should we have two separate ifdef blocks or should the gradient automatically turn on `RENDER_LINE_TRIM_OFFSET` too?

Reply:
You are right, I will change this.

Reply:
After rethinking, I think it needs to be 
```
#if defined(RENDER_LINE_GRADIENT) || defined(RENDER_LINE_TRIM_OFFSET)
```
As only these two require geojson `line-metrics` and a_packed contains all the info relates to it.